mlir-sycl/lib/Transforms/SYCLMethodToSYCLCall.cpp

```diff
@@ -47,62 +45,40 @@ static mlir::Value adaptArgumentForSYCLCall(OpBuilder &Rewriter,
     return Original;
 
   const auto MT = TargetType.cast<MemRefType>();
-  const Type ThisType = Original.getType();
+  const auto ThisType = Original.getType().cast<MemRefType>();
   const llvm::ArrayRef<int64_t> TargetShape = MT.getShape();
   const mlir::Type TargetElementType = MT.getElementType();
   const unsigned TargetMemSpace = MT.getMemorySpaceAsInt();
 
-  const auto CreateAlloca = [=](OpBuilder &Builder) -> memref::AllocaOp {
-    OpBuilder::InsertionGuard Guard{Builder};
-
-    auto ParentFunction = cast<mlir::FunctionOpInterface>(
-        Builder.getInsertionPoint()
-            ->getParentWithTrait<mlir::FunctionOpInterface::Trait>());
-
-    // We first create needed allocas
-    Builder.setInsertionPointToStart(
-        &*ParentFunction.getFunctionBody().getBlocks().begin());
-
-    return Builder.create<memref::AllocaOp>(Loc,
-                                            MemRefType::get({1}, ThisType));
-  };
-
-  auto Alloca = static_cast<Value>(CreateAlloca(Rewriter));
-
-  // Store the element
-  Rewriter.create<memref::StoreOp>(
-      Loc, Original, Alloca,
-      ValueRange{Rewriter.createOrFold<arith::ConstantIndexOp>(Loc, 0)});
-
-  // Cast the memref value to the expected shape
-  Alloca = Rewriter.createOrFold<memref::CastOp>(
-      Loc, MemRefType::get(TargetShape, ThisType), Alloca);
-
-  if (Alloca.getType().cast<MemRefType>().getMemorySpaceAsInt() !=
-      MT.getMemorySpaceAsInt()) {
-    // Cast to the required memspace
-    Alloca = Rewriter.create<polygeist::Memref2PointerOp>(
-        Loc, LLVM::LLVMPointerType::get(ThisType), Alloca);
-    Alloca = Rewriter.create<LLVM::AddrSpaceCastOp>(
-        Loc, LLVM::LLVMPointerType::get(ThisType, TargetMemSpace), Alloca);
-    Alloca = Rewriter.create<polygeist::Pointer2MemrefOp>(
-        Loc, MemRefType::get(TargetShape, ThisType, {}, TargetMemSpace),
-        Alloca);
-  }
-
-  if (Alloca.getType() == MT) {
+  assert(TargetShape == ThisType.getShape() &&
```

Comment:
When I was doing testing for my temporary change, seems like the shape could be different.